internal/provider/helpers.go

```diff
@@ -0,0 +1,256 @@
+package provider
+
+import (
+	"fmt"
+	"strings"
+
+	eboptions "github.com/coder/envbuilder/options"
+	"github.com/coder/serpent"
+	"github.com/coder/terraform-provider-envbuilder/internal/tfutil"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/spf13/pflag"
+)
+
+const (
+	envbuilderOptionPrefix = "ENVBUILDER_"
+)
+
+// nonOverrideOptions are options that cannot be overridden by extra_env.
+var nonOverrideOptions = map[string]struct{}{
+	"ENVBUILDER_CACHE_REPO": {},
+	"ENVBUILDER_GIT_URL":    {},
+}
+
+// optionsFromDataModel converts a CachedImageResourceModel into a corresponding set of
+// Envbuilder options. It returns the options and any diagnostics encountered.
+func optionsFromDataModel(data CachedImageResourceModel) (eboptions.Options, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	var opts eboptions.Options
+
+	// Required options. Cannot be overridden by extra_env.
+	opts.CacheRepo = data.CacheRepo.ValueString()
+	opts.GitURL = data.GitURL.ValueString()
+
+	// Other options can be overridden by extra_env, with a warning.
+	// Keep track of which options are overridden.
+	overrides := make(map[string]struct{})
```

Comment:
Isn't this backwards, these options are _not_ overridden as they're set via params. Extra env produces overrides. Renaming this could make its purpose clearer.

(Also suggestion: boolean map)